app/src/main/java/com/nutomic/syncthingandroid/util/ConfigXml.java

```diff
@@ -313,15 +322,25 @@ private void saveChanges() {
             Log.w(TAG, "Failed to save updated config. Cannot change the owner of the config file.");
             return;
         }
+
+        Log.i(TAG, "Writing updated config file");
+        File mConfigTempFile = Constants.getConfigTempFile(mContext);
         try {
-            Log.i(TAG, "Writing updated config back to file");
             TransformerFactory transformerFactory = TransformerFactory.newInstance();
             Transformer transformer = transformerFactory.newTransformer();
             DOMSource domSource = new DOMSource(mConfig);
-            StreamResult streamResult = new StreamResult(mConfigFile);
+            StreamResult streamResult = new StreamResult(mConfigTempFile);
             transformer.transform(domSource, streamResult);
         } catch (TransformerException e) {
-            Log.w(TAG, "Failed to save updated config", e);
+            Log.w(TAG, "Failed to save temporary config file", e);
+            return;
+        }
+        try {
+            // Delete outdated config file and put temporary config file instead.
+            mConfigFile.delete();
+            mConfigTempFile.renameTo(mConfigFile);
```

Comment:
RenameTo should be enough, no need to delete

Reply:
hmm.. I'll better test this before as JavaDoc states it can be wrong. Link https://docs.oracle.com/javase/6/docs/api/java/io/File.html#renameTo%28java.io.File%29

Reply:
Verified with a debug build and its working correctly without the delete before renameTo. Good point!